### ISSUE TITLE
Ports to be allowed?

### ISSUE THREAD

**Issue by Suncatcher** [aria2/aria2#944]
It is still unclear to me what ports should I allow for smooth Aria2 functioning. I didn't specified the port in config, so it should have been default, 6800. 
However, when I check up listening ports in system, it shows 6800 **and** 6882. And having enabled only 6800 doesn't make Aria2 work! Why?
Moreover, when I enable both 6800 and 6882, the speed is increasingly **LOW**, and I should enable **all ports** to download, which is unacceptable. 
So what concrete ports should I enable in my firewall? And what protocols? TCP or UDP? Or both?

**Comment by tatsuhiro-t** [comment]
Use --listen-port and --dht-listen-port options to specify a single port number to reduce port exposure.

**Comment by Suncatcher** [comment]
And does Aria use TCP or UDP or both?

**Comment by Suncatcher** [comment]
I start Aria like this 
    
    aria2c --enable-rpc --rpc-listen-all --listen-port=6900 --dht-listen-port=6900 --pause-metadata=true --dir=/media/Downloads

But it again listens both for 6800 and 6900. 

![shot-2017-06-19_20-20-26](https://user-images.githubusercontent.com/6388034/27297346-27dd0396-552d-11e7-8a8a-e3cb0e433e01.png)
How to fix this?

**Comment by Suncatcher** [comment]
It **does** download when I enable 6900th port, but **cannot** fetch magnet metadata with only 6900 enabled, so it seems to use another port for fetching torrent metadata. But what?

**Comment by tatsuhiro-t** [comment]
Port 6800 is RPC server port.  If you don't want RPC functionality, use --enable-rpc=false.
Torrent metadata is fetched from DHT network.  aria2 uses port 6900 UDP in your case to do DHT operation.

**Comment by Suncatcher** [comment]
No, I need RPC-server. The problem is in DHT port. Even if I specify 

      --dht-listen-port=6900 

metadata are not retrieved via it, and it works only with disabled firewall. What else ports does it need for DHT?

**Comment by Suncatcher** [comment]
So, RPC-server port is not changeable? Did I understand you correctly? Even if I start Aria with

    --listen-port=6900
it still listen on 6800 port.

**Comment by tatsuhiro-t** [comment]
https://aria2.github.io/manual/en/html/aria2c.html#cmdoption--rpc-listen-port

**Comment by Pound-Hash** [comment]
Does the port specified by the setting `--dht-entry-point` need to be opened in the firewall?
